Lib/test/test_io.py

```diff
@@ -3245,6 +3245,31 @@ def _make_illegal_wrapper():
         t = _make_illegal_wrapper()
         self.assertRaises(TypeError, t.read)
 
+        # Issue 31243: calling read() while the return value of decoder's
+        # getstate() is invalid should neither crash the interpreter nor
+        # raise a SystemError.
+        def _make_very_illegal_wrapper(getstate_ret_val):
+            class BadDecoder:
+                def getstate(self):
+                    return getstate_ret_val
+            class BadIncrementalDecoder:
+                def __call__(self, dummy):
+                    return BadDecoder()
+            quopri = codecs.lookup("quopri")
+            quopri_decoder = quopri.incrementaldecoder
```

Comment:
The `test.support` module is for internal use, and most of its content is not documented. If you want to add a documentation, open a separate issue.